packages/experience-builder-sdk/src/core/editor/EditorModeEntityStore.ts

```diff
@@ -40,18 +40,23 @@ export class EditorModeEntityStore extends EditorEntityStore {
     this.locale = locale;
   }
 
-  async fetchEntities(entityLinks: UnresolvedLink<'Entry' | 'Asset'>[]) {
+  fetchEntities(entityLinks: UnresolvedLink<'Entry' | 'Asset'>[]) {
     const entryLinks = entityLinks.filter((link) => link.sys?.linkType === 'Entry');
     const assetLinks = entityLinks.filter((link) => link.sys?.linkType === 'Asset');
 
-    const uniqueEntryLinks = new Set(entryLinks.map((link) => link.sys.id));
-    const uniqueAssetLinks = new Set(assetLinks.map((link) => link.sys.id));
+    const uniqueEntryIds = [...new Set(entryLinks.map((link) => link.sys.id))];
+    const uniqueAssetIds = [...new Set(assetLinks.map((link) => link.sys.id))];
+
+    const { missing: missingEntryIds } = this.getEntitiesFromMap('Entry', uniqueEntryIds);
+    const { missing: missingAssetIds } = this.getEntitiesFromMap('Asset', uniqueAssetIds);
+
+    // Return false to indicate that no async fetching is happening
+    if (!missingAssetIds.length && !missingEntryIds.length) return false;
 
     // Entries and assets will be stored in entryMap and assetMap
-    await Promise.allSettled([
-      this.fetchEntries([...uniqueEntryLinks]),
-      this.fetchAssets([...uniqueAssetLinks]),
-    ]);
+    return Promise.all([this.fetchEntries(uniqueEntryIds), this.fetchAssets(uniqueAssetIds)]).then(
+      ([entries, assets]) => [...entries, ...assets]
+    );
   }
```

Comment:
The fact that `fetchEntities()` returns strict `false` is a bit surprising (not here, but from the where fetchEntiteis() is called). 

Also the `fetchEntities()` is de-facto an async function (as it returns a promise). What do you think of making it consistent via this snippet:

```ts
  async fetchEntities(entityLinks: UnresolvedLink<'Entry' | 'Asset'>[]) : Promise<false | Array<Entry|Asset>> {
    const entryLinks = entityLinks.filter((link) => link.sys?.linkType === 'Entry');
    const assetLinks = entityLinks.filter((link) => link.sys?.linkType === 'Asset');

    const uniqueEntryIds = [...new Set(entryLinks.map((link) => link.sys.id))];
    const uniqueAssetIds = [...new Set(assetLinks.map((link) => link.sys.id))];

    const { missing: missingEntryIds } = this.getEntitiesFromMap('Entry', uniqueEntryIds);
    const { missing: missingAssetIds } = this.getEntitiesFromMap('Asset', uniqueAssetIds);

    // Return false to indicate that no async fetching is happening
    if (!missingAssetIds.length && !missingEntryIds.length) return false;

    // Entries and assets will be stored in entryMap and assetMap
    return await Promise.all([this.fetchEntries(uniqueEntryIds), this.fetchAssets(uniqueAssetIds)]).then(
      ([entries, assets]) => [...entries, ...assets]
    );
  }
```

Do you think there's a benefit in adding type to the function? 

```
```

Reply:
There is a problem: When I make the `false` response part of the promise. Then I need to "await" it completely. The current logic relies on a synced "false" response. If it's not "false" but a promise, I update the state variable `areEntitiesFetched` to reflect that it just started rerendering. That would not be possible if I transform it like you proposed.
Though, I agree that it's not the most elegant way. I will remember to get back to this if we figure out that the loading bugs are gone. If so, I will try reverting parts of this PR so we have a more beautiful code :)